consumer.go

```diff
@@ -463,6 +471,7 @@ func (w *brokerConsumer) updateSubscriptionCache(newSubscriptions []*partitionCo
 }
 
 func (w *brokerConsumer) abort(err error) {
+	w.consumer.abandonBrokerConsumer(w)
```

Comment:
Another clarification: because we only have one goroutine dealing with this broker, there is no chance of us using the broker's connection between the occurrence of the error, and this call where we abandon it?